recorder/record.py

```diff
@@ -1,105 +1,59 @@
-
-    def open_blob(self, filename):
-        storage_client = storage.Client()
-        bucket = storage_client.bucket(self.BUCKET_NAME)
-        blob = bucket.blob(filename)
-        return blob.open('wb')
-
-    def log_progress(self, filename, elapsed_time):
-        progress = min(round(elapsed_time / self.duration * 100), 100)
-        if math.fmod(progress, 5) == 0 and progress > self.last_progress:
-            logging.info(f'{filename} - {progress}% complete')
-            self.last_progress = progress
-
-    def save_streamed_response(self):
-        start_time = time.time()
-        filename = self.get_filename()
-        file = self.open_blob(filename)
-        chunk_size = 512 * 1024 # 0.5MB
-        attempts = 1
-        recording = True
-
-        while recording:
-            try:
-                logging.info(f'Attempt #{attempts}')
-                with requests.get(self.STREAM_URL, stream=True) as response:
-                    response.raise_for_status()
-                    logging.info(f'{filename} - starting ...')                       
-                    for chunk in response.iter_content(chunk_size=chunk_size):
-                        if chunk:
-                            file.write(chunk)
-                        else:
-                            logging.info('No chunk to write')
-                        
-                        elapsed_time = time.time() - start_time
-                        self.log_progress(filename, elapsed_time)         
-                        if elapsed_time >= self.duration:
-                            recording = False
-                            break
-            except Exception:                
-                logging.exception('Exception occurred')
-                attempts += 1
-                if attempts <= self.MAX_ATTEMPTS:                    
-                    time.sleep(attempts)
-                    continue
-                else:
-                    logging.critical(self.MAX_ATTEMPTS_EXCEEDED)
-                    sys.exit(1)
-
-if __name__ == '__main__':    
-    r = Recorder()
-    r.save_streamed_response()
+import signal
+import google.cloud.logging
+from datetime import datetime
+from google.cloud import storage
+from http.client import HTTPConnection
+from zoneinfo import ZoneInfo
+
+# allll the logging
+HTTPConnection.debuglevel = 1
+logging.basicConfig()
+logging.getLogger().setLevel(logging.DEBUG)
+requests_log = logging.getLogger("urllib3")
+requests_log.setLevel(logging.DEBUG)
+requests_log.propagate = True
+lc = google.cloud.logging.Client()
+lc.setup_logging()
+
+# handle timeouts
+class TimeoutException(Exception):
+    pass
+
+def timeout_handler(signum, frame):
+    raise TimeoutException
+
+signal.signal(signal.SIGALRM, timeout_handler)
+
+
+def record():
+    sc = storage.Client()
+    bucket = sc.bucket(os.getenv('BUCKET_NAME'))
+
+    r = requests.get('http://peridot.streamguys.com:5180/live', stream=True, timeout=25)
```

Comment:
Is there a reason the Stream URL is now hardcoded rather than pulled from the environment?

Reply:
Not a strong one. I started with it hard-coded for simplicity as I started rebuilding, and then didn't need to pull it out to a variable since it doesn't vary across environments.